recipes/spdlog/all/conanfile.py

```diff
@@ -51,98 +50,104 @@ def configure(self):
             del self.options.fPIC
 
     def requirements(self):
-        if tools.Version(self.version) >= "1.10.0":
-            self.requires("fmt/8.1.1")
-        elif tools.Version(self.version) >= "1.9.0":
-            self.requires("fmt/8.0.1")
-        elif tools.Version(self.version) >= "1.7.0":
-            self.requires("fmt/7.1.3")
-        elif tools.Version(self.version) >= "1.5.0":
-            self.requires("fmt/6.2.1")
+        # TODO: Remove in Conan 2.0 - 1.x self.requires does not support transitive_headers
+        requires = lambda ref: self.requires(ref, transitive_headers=True) if Version(conan_version) >= "2.0.0-beta" else self.requires(ref)
```

Comment:
Indeed spdlog needs fmt, but the main problem is, spdlog includes fmt/core.h in a public and exported header, so when you consume spdlog.h, it will ask for fmt directly too. That's why it need to be transitive. I talked with Carlos about this topic, he said it's a rare case. Indeed it should be, because we usually don't expose external dependencies as a good practice. Do you have another suggestion related to the package ID?

Reply:
To be more clear about the dependency chain: https://github.com/gabime/spdlog/blob/v1.10.0/include/spdlog/spdlog.h#L12 -> https://github.com/gabime/spdlog/blob/v1.10.0/include/spdlog/common.h#L45 -> https://github.com/gabime/spdlog/blob/v1.10.0/include/spdlog/fmt/fmt.h#L27

Reply:
BTW I commented about the case with Conan team here: https://github.com/conan-io/conan/issues/11760